File: Hide.java
// Hide.java
class Homer{
	void doh(char c){
		System.out.println("doh(char)");
	}
	void doh(float f){
		System.out.println("doh(float)");
	}
}

class Milhouse{}

class Bart extends Homer{
	/* 如果使用这个注解，编译时会报错：
	*  “方法不会覆盖或实现超类型的方法” -- method does not override a method from its superclass
	*/
	//@Override 
	void doh(Milhouse m){
		System.out.println("doh(Milhouse)");
	}
}

public class Hide{
	public static void main(String[] args){
		Bart b = new Bart();
		b.doh('x');
		b.doh(1.0f);
		b.doh(new Milhouse());
	}
}
/* Output:

*/

File: Hide.cpp
// Hide.cpp
// #include "Hide.h"
#include <iostream>
using namespace::std;

// Hide.h start
class Homer {
public:
	Homer();
	~Homer();
	void doh(char);
	void doh(float);
};
class Milhouse {
public:
	Milhouse();
	~Milhouse();
};
class Bart : public Homer {
public:
	Bart();
	~Bart();
	void doh(Milhouse*);
};
// Hide.h end

Homer::Homer(){}
Homer::~Homer(){}
void Homer::doh(char c){
	cout << "doh(char)" << endl;
}
void Homer::doh(float f){
	cout << "doh(float)" << endl;
}

Milhouse::Milhouse(){}
Milhouse::~Milhouse(){}

Bart::Bart(){
	Homer();
}
Bart::~Bart(){}
void Bart::doh(Milhouse* m){
	cout << "doh(Milhouse)" << endl;
}

int main(int argc, char* argv[]){
	Bart* b = new Bart();
	//b->doh('x'); // error C2664: 'void Bart::doh(Milhouse *)': cannot convert argument 1 from 'char' to 'Milhouse *'
	//b->doh(1.0f); // error C2664: 'void Bart::doh(Milhouse *)': cannot convert argument 1 from 'float' to 'Milhouse *'
	b->doh(new Milhouse());
	return 0;
}
/* Output:

*/